crates/lib/src/action/workingdata.rs

```diff
@@ -176,15 +179,26 @@ impl Action {
 		guard.insert(process, (Resolution::Apply(outcome), set));
 	}
 
-	/// Returns a snapshot of the `SupervisorId`s of the running `Process`es at creation of the
+	/// Returns a snapshot of the [`SupervisorId`]s of the running [`Command`]s at creation of the
 	/// [`Action`].
 	pub fn current_processes(&self) -> &[SupervisorId] {
```

Comment:
should also be renamed away from process, and its comment maybe changed to "...commands alive at the creation time..." because I think it could be a bit confusing, ie one could think that means only the commands currently actively running (but it would also include supervisors for stopped commands).

Reply:
oh, yess, that one was silly to miss! 

yes, true, that does have a very different implication from what is actually happening! good catch! do stopped `Supervisor`s get cleaned up at some point, or is the machinery that would allow the user to do so?

Reply:
Hmm yeah I think we should either:
- add an outcome called `Outcome::End` that does this, and gets called from on_process, or
- more explicit, add a `remove_command()` method here

Reply:
I'm leaning more to the second actually

Reply:
yess, both would be nice, but second one does maybe state it's purpose a bit more explicitly! it would probably also help with the discoverability of this *removal* feature ! 

either way, should we let the `Supervisor` know this is going on, or would we simply `Outcome::Stop` the `OutcomeWorker` and then remove the command from the `HashMap` that tracks the alive commands?

Reply:
> would we simply Outcome::Stop the OutcomeWorker and then remove the command from the HashMap that tracks the alive commands?

Hmm, I think that approach sounds fine yes.

Reply:
one question then, though maybe silly! should stopping a `Command` that does not yet exist in the alive `Command` set that is returned from `current_commands()` do anything? that is, even if a `Command` does exist, because it was started from the same call to `action_handler::handle()` result in something happening, or should  this be filtered out?

Reply:
Hmm. I think:
- if the command exists only in the action (ie it was just created), then stopping it should remove it from there
- if the command doesn't exist at all, then stopping it should silently do nothing